src/lsdb/catalog/margin_catalog.py

```diff
@@ -23,3 +25,20 @@ def __init__(
         hc_structure: hc.catalog.MarginCatalog,
     ):
         super().__init__(ddf, ddf_pixel_map, hc_structure)
+
+
+def _validate_margin_catalog(margin_hc_catalog, hc_catalog):
+    """Validate that the margin and main catalogs have compatible schemas. The order of
+    the pyarrow fields should not matter."""
+    expected_margin_schema = _create_margin_schema(hc_catalog.schema)
+    # Compare the field set for the schemas. They should match.
+    if set(margin_hc_catalog.schema) != set(expected_margin_schema):
```

Comment:
I don't like this set comparison because duplicated columns won't be detected. Either do a list and sort them then compare, or check the length of the sets are equal to the length of the schema

Reply:
Actually, the length thing won't work because we want to allow duplicates if both schemas have them